supertokens_python/recipe/emailpassword/utils.py

```diff
@@ -302,14 +296,34 @@ def validate_and_normalise_user_input(recipe: EmailPasswordRecipe, app_info: App
                                       reset_password_using_token_feature: Union[
                                           InputResetPasswordUsingTokenFeature, None] = None,
                                       email_verification_feature: Union[InputEmailVerificationConfig, None] = None,
-                                      override: Union[InputOverrideConfig, None] = None) -> EmailPasswordConfig:
+                                      override: Union[InputOverrideConfig, None] = None,
+                                      email_delivery_config: Union[EmailDeliveryConfig[TypeEmailPasswordEmailDeliveryInput], None] = None
+                                      ) -> EmailPasswordConfig:
     if override is None:
         override = InputOverrideConfig()
     if reset_password_using_token_feature is None:
         reset_password_using_token_feature = InputResetPasswordUsingTokenFeature()
 
     if sign_up_feature is None:
         sign_up_feature = InputSignUpFeature()
+
+    def get_email_delivery_config(
+        ep_recipe: RecipeImplementation,
+    ) -> EmailDeliveryConfigWithService[TypeEmailPasswordEmailDeliveryInput]:
+        if email_delivery_config and email_delivery_config.service:
+            return EmailDeliveryConfigWithService(
+                service=email_delivery_config.service,
+                override=email_delivery_config.override
+            )
+
+        email_service = BackwardCompatibilityService(
+            app_info=app_info,
+            recipe_interface_impl=ep_recipe,
+            reset_password_using_token_feature=reset_password_using_token_feature,
+            email_verification_feature=email_verification_feature,
+        )
+        return EmailDeliveryConfigWithService(email_service, override=None)
```

Comment:
You should not give override as None! Give it as what the user may have passed in `email_delivery_config`.

You need to add tests for this

Reply:
Fixed. Will add tests.